Sources/PackageCollections/Providers/GitHubPackageMetadataProvider.swift

```diff
@@ -144,22 +154,12 @@ struct GitHubPackageMetadataProvider: PackageMetadataProvider {
                         processedAt: Date()
                     )
 
-                    if let cache = self.cache {
-                        cache[reference] = (model, DispatchTime.now())
-
-                        if cache.count > self.configuration.cacheSize {
```

Comment:
With the max size feature in SQLite cache, we don't need to manually delete cache entries.